kubernetes/adapter.go

```diff
@@ -291,26 +272,10 @@ func newIngressForKube(i *Ingress) *ingress {
 }
 
 func newRouteGroupForKube(i *Ingress) *routegroup {
-	shared := "true"
-
-	if !i.Shared {
-		shared = "false"
-	}
+	ing := newIngressForKube(i)
```

Comment:
this reads a bit weird, maybe better to refactor it into it's own function to prepare the Metadata.
wdyt?

Reply:
Moved the metadata creation into a new `newMetadataForKube` function